repository_service_tuf_worker/repository.py

```diff
@@ -535,24 +539,28 @@ def publish_targets(self):
                     new_snapshot_meta.append((rolename, role.signed.version))
 
                 # context lock finished
-                lock_finished = True
+                lock_status_targets = True
 
             # update snapshot and timestamp
             # note: the `db_published_targets` contains the targets that
             # needs to updated in SQL DB as 'published' and it will be done
             # by the `_update_timestamp`
-            self._update_timestamp(
-                self._update_snapshot(new_snapshot_meta),
-                db_published_targets,
-            )
+            status_lock_timestamp = False
+            with self._redis.lock(LOCK_TIMESTAMP, timeout=timeout):
```

Comment:
Can you add a comment explaining why this is required? 
You can mention that the automatic bump online roles task will try to update `timestamp` (and `snapshot`) while you are updating it here.